packages/graph/hash_graph/lib/graph/src/types/schema/array.rs

```diff
@@ -0,0 +1,313 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum ArrayTypeTag {
+    Array,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct Array {
+    r#type: ArrayTypeTag,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    min_items: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    max_items: Option<usize>,
+}
+
+impl Array {
+    #[must_use]
+    pub fn new(min_items: impl Into<Option<usize>>, max_items: impl Into<Option<usize>>) -> Self {
+        Self {
+            r#type: ArrayTypeTag::Array,
+            min_items: min_items.into(),
+            max_items: max_items.into(),
+        }
+    }
+
+    #[must_use]
+    pub const fn min_items(&self) -> Option<usize> {
+        self.min_items
+    }
+
+    #[must_use]
+    pub const fn max_items(&self) -> Option<usize> {
+        self.max_items
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct TypedArray<T> {
+    items: T,
+    #[serde(flatten)]
+    array: Array,
+}
+
+impl<T> TypedArray<T> {
+    #[must_use]
+    pub fn new(
+        items: T,
+        min_items: impl Into<Option<usize>>,
+        max_items: impl Into<Option<usize>>,
+    ) -> Self {
+        Self {
+            items,
+            array: Array::new(min_items, max_items),
+        }
+    }
+
+    #[must_use]
+    pub const fn items(&self) -> &T {
+        &self.items
+    }
+
+    #[must_use]
+    pub const fn min_items(&self) -> Option<usize> {
+        self.array.min_items()
+    }
+
+    #[must_use]
+    pub const fn max_items(&self) -> Option<usize> {
+        self.array.max_items()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct MaybeOrderedArray {
+    #[serde(default)]
```

Comment:
Can we perhaps explicitly mention the default? Might be more likely to update it if we ever change the decision of true/false.

Reply:
It's not easily possible to specify `default = "false"`. It would require an additional function `fn default_false() -> bool { false }`. To clarify this I added a comment. Do you think it's worth adding a function? 3764bdf092c2f844eefd5f62dfdb6af6954b32bf

Reply:
Ah, probably not worth a function then. Thanks for the clarification